eva/optimizer/rules/rules.py

```diff
@@ -404,6 +406,37 @@ def apply(self, before: LogicalFilter, context: OptimizerContext):
         yield root_node
 
 
+class XformExtractObjectToLinearFlow(Rule):
+    """If the inner node of a lateral join is a Extract_Object function-valued
+    expression, we eliminate the join node and make the inner node the parent of the
+    outer node. This produces a linear data flow path."""
+
+    #     LogicalJoin(Lateral)               LogicalExtractObject
+    #     /           \                 ->        |
+    #    A        LogicalExtractObject            A
+
+    def __init__(self):
```

Comment:
This is a special case. Ideally, I think it will be better to we can base UDFs on `LogicalApplyAndMerge`. So later, if we have a rule for ray, or for the old reuse, or any function-oriented optimization. We do not need to consider this as a special case and handle it accordingly.

Reply:
I see. Reduces redundancy. Good idea!

Reply:
Thanks for this idea! The logic is quite clean now.